src/net.cpp

```diff
@@ -1906,6 +1979,20 @@ void CConnman::DisconnectNodes()
                 // remove from m_nodes
                 m_nodes.erase(remove(m_nodes.begin(), m_nodes.end(), pnode), m_nodes.end());
 
+                // Add to reconnection list if appropriate. We don't reconnect right here, because
+                // the creation of a connection is a blocking operation (up to several seconds),
+                // and we don't want to hold up the socket handler thread for that long.
+                if (pnode->m_transport->ShouldReconnectV1()) {
```

Comment:
When we fail to connect using V2 and succeed to connect using V1, should we bother removing `NODE_P2P_V2` for that node from addrman?

Reply:
> If a peer lies about their supported services, they only hurt their own ability of being connected to

Peers could lie about each-other in gossip. But it doesn't matter here, as you say.

Reply:
Worth noting here is that if an address is already in addrman and we receive it via gossip we update its services, but in a cumulative manner - only add new services, without removing existent services from addrman:

https://github.com/bitcoin/bitcoin/blob/dcf6230f92d491f46d2bf6cfc096ab5874e385c9/src/addrman.cpp#L572